cloudos_cli/clos.py

```diff
@@ -944,24 +945,83 @@ def get_workflow_content(self, workspace_id, workflow_name, verify=True):
         }
+        # determine pagination, there might be a lot with the same name
         url = f"{self.cloudos_url}/api/v3/workflows?teamId={workspace_id}&search={workflow_name}"
         response = retry_requests_get(url, headers=headers, verify=verify)
         if response.status_code >= 400:
             raise BadRequestException(response)
-        content = json.loads(response.content)
-        return content
+        pag_content = json.loads(response.content)
+        max_pagination = pag_content["paginationMetadata"]["Pagination-Count"]
+
+        return max_pagination
 
-    def workflow_content_query(self, workspace_id, workflow_name, verify=True, query="workflowType"):
+    def get_workflow_content(self, workspace_id, workflow_name, verify=True, last=False, max_page_size=100):
+        """Retrieve the workflow content from API.
+
+        Parameters
+        ----------
+        workspace_id : str
+            The CloudOS workspace ID to search for the workflow.
+        workflow_name : str
+            The name of the workflow to search for.
+        verify : [bool | str], optional
+            Whether to use SSL verification or not. Alternatively, if
+            a string is passed, it will be interpreted as the path to
+            the SSL certificate file. Default is True.
 
-        content = self.get_workflow_content(workspace_id, workflow_name, verify=verify)
+        Returns
+        -------
+        dict
+            The server response containing workflow details.
+
+        Raises
+        ------
+        BadRequestException
+            If the request to retrieve the project fails with a status code
+            indicating an error.
+        """
+        headers = {
+            "Content-type": "application/json",
+            "apikey": self.apikey
+        }
+        max_pagination = self.get_workflow_max_pagination(workspace_id, workflow_name, verify=verify)
+
+        # get all the matching content
+        if max_pagination > max_page_size:
+            content = {"workflows": []}
+            for page_start in range(0, max_pagination, max_page_size):
+                page_size = min(max_page_size, max_pagination - page_start)
+                url = f"{self.cloudos_url}/api/v3/workflows?teamId={workspace_id}&search={workflow_name}&pageSize={page_size}&page={page_start // max_page_size + 1}"
+                response = retry_requests_get(url, headers=headers, verify=verify)
+                # keep structure as a dict
+                content["workflows"].extend(json.loads(response.content).get("workflows", []))
+        else:
+            url = f"{self.cloudos_url}/api/v3/workflows?teamId={workspace_id}&search={workflow_name}&pageSize={max_pagination}"
+            response = retry_requests_get(url, headers=headers, verify=verify)
+            # return all content
+            content = json.loads(response.content)
+
+        if response.status_code >= 400:
+            raise BadRequestException(response)
```

Comment:
```suggestion
        if response.status_code > 400:
            raise BadRequestException(response)
```
Very surprisingly, when you search for a non-existent workflow (`cloudos job run --workflow-name non_existent_workflow -p a=1`), even when the response is in fact a dict with an empty list of workflows, the `response.status_code` is `400`, making the error less informative.